src/main/java/net/minestom/server/entity/Entity.java

```diff
@@ -292,6 +293,12 @@ public boolean isOnGround() {
      */
     public @NotNull CompletableFuture<Void> teleport(@NotNull Pos position, long @Nullable [] chunks) {
         Check.stateCondition(instance == null, "You need to use Entity#setInstance before teleporting an entity!");
+        
+        EntityTeleportEvent teleportEvent = this instanceof Player p ? new PlayerTeleportEvent(p, position) : new EntityTeleportEvent(this, position);
+        EventDispatcher.call(teleportEvent);
```

Comment:
You may want to replace this with `EventDispatcher.callCancellable()` instead, may make the code a bit cleaner.

Reply:
`EventDispatcher.callCancellable()` can't return a value. For me, it's not the best way for this method, and I made the same as another method below